packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts

```diff
@@ -269,259 +548,255 @@ export class GaussianSplattingMesh extends Mesh {
+                    break;
+                case PLYValue.SCALE_0:
+                    scale[0] = Math.exp(value);
+                    break;
+                case PLYValue.SCALE_1:
+                    scale[1] = Math.exp(value);
+                    break;
+                case PLYValue.SCALE_2:
+                    scale[2] = Math.exp(value);
+                    break;
+                case PLYValue.DIFFUSE_RED:
+                    rgba[0] = value;
+                    break;
+                case PLYValue.DIFFUSE_GREEN:
+                    rgba[1] = value;
+                    break;
+                case PLYValue.DIFFUSE_BLUE:
+                    rgba[2] = value;
+                    break;
+                case PLYValue.F_DC_0:
+                    rgba[0] = (0.5 + GaussianSplattingMesh._SH_C0 * value) * 255;
+                    break;
+                case PLYValue.F_DC_1:
+                    rgba[1] = (0.5 + GaussianSplattingMesh._SH_C0 * value) * 255;
+                    break;
+                case PLYValue.F_DC_2:
+                    rgba[2] = (0.5 + GaussianSplattingMesh._SH_C0 * value) * 255;
+                    break;
+                case PLYValue.F_DC_3:
+                    rgba[3] = (0.5 + GaussianSplattingMesh._SH_C0 * value) * 255;
+                    break;
+                case PLYValue.OPACITY:
+                    rgba[3] = (1 / (1 + Math.exp(-value))) * 255;
+                    break;
+                case PLYValue.ROT_0:
+                    r0 = value;
+                    break;
+                case PLYValue.ROT_1:
+                    r1 = value;
+                    break;
+                case PLYValue.ROT_2:
+                    r2 = value;
+                    break;
+                case PLYValue.ROT_3:
+                    r3 = value;
+                    break;
+            }
         }
 
-        return buffer;
+        q.set(r1, r2, r3, r0);
+        q.normalize();
+        rot[0] = q.w * 128 + 128;
+        rot[1] = q.x * 128 + 128;
+        rot[2] = q.y * 128 + 128;
+        rot[3] = q.z * 128 + 128;
+        offset.value += header.rowVertexLength;
+    }
+
+    /**
+     * Converts a .ply data array buffer to splat
+     * if data array buffer is not ply, returns the original buffer
+     * @param data the .ply data to load
+     * @param useCoroutine use coroutine and yield
+     * @returns the loaded splat buffer
+     */
+    public static *ConvertPLYToSplat(data: ArrayBuffer, useCoroutine = false) {
+        const header = GaussianSplattingMesh.ParseHeader(data);
+        if (!header) {
+            return data;
+        }
+
+        const offset = { value: 0 };
+        const compressedChunks = GaussianSplattingMesh._GetCompressedChunks(header, offset);
+
+        for (let i = 0; i < header.vertexCount; i++) {
+            GaussianSplattingMesh._GetSplat(header, i, compressedChunks, offset);
+            if (i % 30000 === 0 && useCoroutine) {
+                yield;
+            }

```

Comment:
Is it too much overhead of if you remove the `i % 30000 === 0` part of the check (since the yielding scheduler will async yield based on elapsed time)?

Reply:
Yes, there is a big overhead. no if: 3s for a 3M splat compressed ply. with if: 2s, same ply. I did 20 runs per test. numbers are pretty stable in both cases.